stdlib/public/Differentiation/FloatingPointDifferentiation.swift.gyb

```diff
@@ -54,205 +48,246 @@ extension ${Self}: Differentiable {
+%  linear_map_kind = 'differential' if derivative_kind == 'jvp' else 'pullback'
+  // TODO: Add derivative of `init<Source>()` where Source: FloatingPoint
+  // TODO: Add derivative of `init(signOf:magnitudeOf:)`
+  // TODO: Add derivative of `sign`
+  // TODO: Add derivative of `magnitude`
+  // TODO: Add derivative of `round()`
+  // TODO: Add derivative of `rounded()`
+
+  @inlinable
   @derivative(of: -)
-  static func _vjpNegate(x: ${Self})
-    -> (value: ${Self}, pullback: (${Self}) -> ${Self}) {
+  static func _${derivative_kind}Negate(
+    _ x: ${Self}
+  ) -> (value: ${Self}, ${linear_map_kind}: (${Self}) -> ${Self}) {
     return (-x, { v in -v })
   }
 
-  @usableFromInline
-  @_transparent
-  @derivative(of: -)
-  static func _jvpNegate(x: ${Self})
-  -> (value: ${Self}, differential: (${Self}) -> ${Self}) {
-    return (-x, { dx in -dx })
+  @inlinable
+  @derivative(of: negate)
+  mutating func _${derivative_kind}NegateAssign() -> (
+    value: Void, ${linear_map_kind}: (inout Self) -> Void
+  ) {
+    negate()
+    return ((), { dx in dx.negate() })
   }
+
+  @inlinable
+  @derivative(of: formSquareRoot)
+  mutating func _${derivative_kind}FormSquareRoot() -> (
+    value: Void, ${linear_map_kind}: (inout Self) -> Void
+  ) {
+    formSquareRoot()
+    let value = self
+    return ((), { v in v /= 2 * value })
+  }
+%end # for derivative_kind in ['jvp', 'vjp']:
 }
 
 /// Derivatives of standard binary operators.
 ${Availability(bits)}
 extension ${Self} {
-  ${Availability(bits)}
-  @inlinable // FIXME(sil-serialize-all)
-  @_transparent
+  // TODO: Add derivative of `formRemainder(dividingBy:)`
+  // TODO: Add derivative of `formTruncatingRemainder(dividingBy:)`
+  // TODO: Add derivative of `remainder(dividingBy:)`
+  // TODO: Add derivative of `truncatingRemainder(dividingBy:)`
+
+  @inlinable
   @derivative(of: +)
   static func _vjpAdd(
-    lhs: ${Self}, rhs: ${Self}
+    _ lhs: ${Self}, 
+    _ rhs: ${Self}
   ) -> (value: ${Self}, pullback: (${Self}) -> (${Self}, ${Self})) {
     return (lhs + rhs, { v in (v, v) })
   }
 
-  ${Availability(bits)}
-  @inlinable // FIXME(sil-serialize-all)
-  @_transparent
+  @inlinable
   @derivative(of: +)
   static func _jvpAdd(
-    lhs: ${Self}, rhs: ${Self}
+    _ lhs: ${Self}, 
+    _ rhs: ${Self}
   ) -> (value: ${Self}, differential: (${Self}, ${Self}) -> ${Self}) {
     return (lhs + rhs, { (dlhs, drhs) in dlhs + drhs })
   }
 
-  ${Availability(bits)}
-  @inlinable // FIXME(sil-serialize-all)
-  @_transparent
+  @inlinable
   @derivative(of: +=)
-  static func _vjpAddAssign(_ lhs: inout ${Self}, _ rhs: ${Self}) -> (
-    value: Void, pullback: (inout ${Self}) -> ${Self}
-  ) {
+  static func _vjpAddAssign(
+    _ lhs: inout ${Self}, 
+    _ rhs: ${Self}
+  ) -> (value: Void, pullback: (inout ${Self}) -> ${Self}) {
     lhs += rhs
     return ((), { v in v })
   }
 
-  ${Availability(bits)}
-  @inlinable // FIXME(sil-serialize-all)
-  @_transparent
+  @inlinable
```

Comment:
Why did the transparency attributes come off these declarations?

Reply:
I don't know why it was there to begin with.

Reply:
`@_transparent` forces inlining to occur at `-Onone`. `@inlinable` merely makes the definition of the associated function available to the inline, it does not make it mandatory.

Reply:
This file hasn't been well-maintained for 2 years. There was even a function with an incorrect derivative that slipped by tests. I see no good justification for keeping @_transparent on just a portion of these functions.